api-report/tree.api.md

```diff
@@ -233,19 +233,23 @@ export interface DetachedField extends Opaque<Brand<string, "tree.DetachedField"
 const DUMMY_INVERT_TAG: ChangesetTag_2;
 
 // @public
-export interface EditableTree {
+export type EditableField = readonly [FieldSchema, FieldKey, readonly EditableTree[]];
+
+// @public
+export interface EditableTree extends Iterable<EditableField> {
     readonly [anchorSymbol]: Anchor;
     readonly [getTypeSymbol]: (key?: FieldKey, nameOnly?: boolean) => NamedTreeSchema | TreeSchemaIdentifier | undefined;
     readonly [proxyTargetSymbol]: object;
     readonly [valueSymbol]: Value;
-    readonly [key: FieldKey]: UnwrappedEditableField;
+    readonly [key: FieldKey]: UnwrappedEditableField | EditableField;
```

Comment:
If someone calls this and gets an array back, how are they supposed to know if the array is an array of children, or an EditableField array?

I think it makes sense to leave this return type as it was, and only use EditableField for APIs where the code receiving the value didn't just specify the key (So just use it when iterating for now).

This would also kelp keep the changes in this PR focused on one thing (allowing iteration).

Reply:
Indeed, I was not sure about this, as for me it seems natural to have an iterator iterating over the same type as the field itself, but if you think it's fine, it's also fine for me, makes things simpler.

So, removed.